### DISCUSSION TITLE
Consider doing "-rc" releases on PyPI

### DISCUSSION THREAD

**Discussion by cbm755** [pymupdf/PyMuPDF#2359] [Ideas]
I think before 1.22.0, there might've been a `1.22.0-rc` (?)   At least I think I saw references to  that in the docs.

But I don't think it ever appeared on PyPI.  It seems it is possible to "tag" a pre-release on PyPI for wider testing.  Maybe that would be a good idea in the future?

Here's an example for Pandas: https://pypi.org/project/pandas/#history

- - - -

I think (but haven't tried it myself) that users would then be able to install  with `pip install --pre pymupdf`

**Comment by cbm755** [comment]
oh, in fact this was done in the past: https://pypi.org/project/PyMuPDF/#history

In that case, it was likely just overlooked for the latest release: maybe this should be bug, not an request for enhancement?

**Comment by julian-smith-artifex-com** [comment]
We decided not to use release candidates for 1.22.0. Please open a Discussion if you want to talk more about this.

**Comment by JorjMcKie** [comment]
I allow myself to add that a qualification as "bug" would be pointless, because it were impossible to fix it.

**Comment by cbm755** [comment]
@JorjMcKie I did mean "for future releases", not backwards in time ;-)

@julian-smith-artifex-com Sure, we can move it to discussion: I don't see button to do that, perhaps you can?

My $0.02 for that discussion: if you'd had a -rc, I might've run my test suite against it and we'd have caught #2348 before release.  (Of course I might not have gotten around to it!  But perhaps I can tool my CI to run a `pip install --pre pymupdf` job for example.)

**Comment by cbm755** [comment]
@JorjMcKie "bug" in the sense of "Your CI/CD for 'rc' tags is not pushing to PyPI".

**Comment by julian-smith-artifex-com** [comment]
Release candidates can certainly be useful but they are not always the best way of doing things.

For example i'd prefer to add extra tests to PyMuPDF's test suite, and generally improve the test suite's test coverage, rather than create release-candidates in the hope that external test suites with extra tests will test them and report back.

That way we can gradually move towards having the code in git being always releasable. For example we already use Github actions to run the full test suite every day, albeit not currently on all platforms.

Are your PyMuPDF-related tests generally available? If so i'd be interested in integrating them into PyMuPDF's test suite.
